_posts/2020-07-13-newpipe-0.21.6-released.md

```diff
@@ -0,0 +1,105 @@
+
+<!-- more -->
+
+## New
+
+You can now tap on hashtags in the stream description, which will execute a search with that hashtag. This was added by [@TiA4f8R](https://github.com/TiA4f8R).
+
+[@mauriciocolli](https://github.com/mauriciocolli) introduced a new button which lets you hide videos you've already played within a feed! Simple in theory, but quite complex in execution. Eventually, [@Stypox](https://github.com/Stypox) took over the PR and finished ironing out the kinks, so enjoy your neat feeds devoid of previously seen videos!
+
+[@Dakkaron](https://github.com/Dakkaron) added a new <code>Appearance</code> setting to manually enable or disable Tablet Mode. So if you have an Android TV which NewPipe mistakenly considers a phone or vice-versa, you can fiddle with this setting, so it works to your liking.
+
+### Improved
+
+Full SAF ([Storage Access Framework](https://developer.android.com/guide/topics/providers/document-provider)) support is finally here! This makes NewPipe future-proof with respect to data storage and retrieval. Why? In Android 10, Google introduced a security feature known as Scoped Storage, which only lets an app see the specific folder or folders in the filesystem which pertain to it, and nothing else (such as your private data from another app). This requires SAF to be enabled. If that wasn't enough, SAF enables storing your downloads on external storage! This massive undertaking was initiated over a year ago by [@wb9688](https://github.com/wb9688), and was then taken on by [@Stypox](https://github.com/Stypox), who improved upon it. Thanks to both of them for getting this implemented! SAF will automatically be enabled for you if you have Android 6 and above, and is now the only option on Android 10 and above. Meanwhile, Kitkat users can only use the native file picker. Note: SAF is still disabled on Fire TV due to a Fire OS GUI bug.
+
+NewPipe has learned to detect deleted channels, both upon opening them from the Subscriptions list, and when refreshing a feed containing such channels! Upon encountering such channels during refresh, the app will notify you and prompt you to delete them. This functionality required a small addition to NewPipe Extractor as well. [@TobiGr](https://github.com/TobiGr) took care of both sides, with help from [@Stypox](https://github.com/Stypox) regarding implementation. So no more snackbar errors every time you update a feed which has deleted channels! Moreover, opening a terminated channel will show you a proper error message explaining why the channel was terminated, instead of a generic error.
```

Comment:
```suggestion
NewPipe has learned to detect deleted channels, both upon opening them from the Subscriptions list, and when refreshing a feed containing such channels! Upon encountering such channels during  a refresh, the app will notify you and prompt you to delete them. This functionality required a small addition to NewPipe Extractor as well. [@TobiGr](https://github.com/TobiGr) took care of both sides, with help from [@Stypox](https://github.com/Stypox) regarding implementation. So no more snackbar errors every time you update a feed which has deleted channels! Moreover, opening a terminated channel will show you a proper error message explaining why the channel was terminated, instead of a generic error.
```